esiclient/v1/trunk.py

```diff
@@ -0,0 +1,142 @@
+                         "\n".join(port_names),
+                         "\n".join(network_names)])
+
+        return ["Trunk", "Port", "Network"], data
+
+
+class Create(command.ShowOne):
+    """Create trunk port with subports"""
+
+    log = logging.getLogger(__name__ + ".Create")
+
+    def get_parser(self, prog_name):
+        parser = super(Create, self).get_parser(prog_name)
+        parser.add_argument(
+            "name",
+            metavar="<name>",
+            help=_("Name of trunk"))
+        parser.add_argument(
+            "--native-network",
+            metavar="<native_network>",
+            help=_("Name or UUID of the native network"))
+        parser.add_argument(
+            '--tagged-networks',
+            default=[],
+            dest='tagged_networks',
+            action='append',
+            metavar='<tagged_networks',
+            help=_("Name or UUID of tagged network")
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)" % parsed_args)
+
+        neutron_client = self.app.client_manager.network
+
+        trunk_name = parsed_args.name
+        network = neutron_client.find_network(parsed_args.native_network)
+        tagged_networks = parsed_args.tagged_networks
+
+        trunk_port = neutron_client.create_port(
+            name="{0}-{1}-trunk-port".format(trunk_name, network.name),
+            network_id=network.id
+        )
+
+        sub_ports = []
+        for tagged_network_name in tagged_networks:
+            tagged_network = neutron_client.find_network(
+                tagged_network_name)
+            sub_port = neutron_client.create_port(
+                name="{0}-{1}-sub-port".format(trunk_name,
+                                               tagged_network.name),
+                network_id=tagged_network.id
+            )
+            sub_ports.append({
+                'port_id': sub_port.id,
+                'segmentation_type': 'vlan',
+                'segmentation_id': tagged_network.provider_segmentation_id
+            })
+
+        trunk = neutron_client.create_trunk(
+            name=trunk_name,
+            port_id=trunk_port.id,
+            sub_ports=sub_ports
+        )
+
+        return ["Trunk", "Port", "Sub Ports"], \
+            [trunk.name,
+             trunk_port.name,
+             trunk.sub_ports]
+
+
+class Delete(command.Command):
+    """Delete trunk port and subports"""
+
+    log = logging.getLogger(__name__ + ".Delete")
+
+    def get_parser(self, prog_name):
+        parser = super(Delete, self).get_parser(prog_name)
+        parser.add_argument(
+            "name",
+            metavar="<name>",
+            help=_("Name of trunk"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)" % parsed_args)
+
+        neutron_client = self.app.client_manager.network
+        trunk = neutron_client.find_trunk(parsed_args.name)
```

Comment:
If the argument is a non-exist trunk name, should it return directly?

Reply:
Ah, my thought was that it would just return whatever error Neutron would return if the trunk doesn't exist. Do you think I should be more explicit?

Reply:
I might suggest that because I don't know what kind of error Neutron would raise. Also, in creation, is it better to check if the trunk (name) already exists before creating a trunk twice?

Reply:
I think in general, my philosophy here is that we're just calling OpenStack commands and allowing whatever behavior that those OpenStack commands allow; we're not trying to override their capabilities. This way, if OpenStack functionality changes, we change right along with it. Let me know if you disagree!

Reply:
It makes sense to me now. We're calling OpenStack commands rather than creating new commands. Thanks for the explanation!